crates/trie/sparse-parallel/src/trie.rs

```diff
@@ -92,6 +92,50 @@ impl ParallelSparseTrie {
         }
     }
 
+    /// Gets or creates a lower subtrie at the given index, updating its path if the provided path
+    /// is shorter.
+    ///
+    /// This method is required for correctly handling extension nodes that cross the upper/lower
+    /// trie boundary. Unlike `lower_subtrie_for_path` which always creates subtries with paths
+    /// truncated to `UPPER_TRIE_MAX_DEPTH` (2 nibbles), this method allows subtries to have
+    /// a path field which is consistent with the its root branch node.
+    ///
+    /// This method is meant to be used when we will be adding nodes directly to the trie.
+    ///
+    /// When an extension node in the upper trie extends beyond 2 nibbles (e.g., 0x123), and we
+    /// need to move nodes to the lower subtrie, the subtrie must be created with the full
+    /// extension path (0x123), not just the first 2 nibbles (0x12). Otherwise, it's not possible
+    /// to easily traverse the lower subtrie, since the subtrie root is not easily identifiable
+    /// without the path field being set correctly.
+    ///
+    /// ## Example:
+    ///
+    /// If we have an extension node 0x → 0x123 in the upper trie, and we're inserting nodes
+    /// at 0x1234, 0x1235, and 0x1236:
+    /// - Incorrect: Create subtrie with path 0x12. subtrie.update_leaf would not be able to
+    /// traverse here because it starts at the subtrie path.
+    /// - Correct: Create subtrie with path 0x123. Here subtrie.update_leaf would start at 0x123,
+    /// correctly find the branch node, and proceed normally.
+    ///
+    /// The method also handles path updates: if a subtrie exists with path 0x12345 and we
+    /// insert a node at 0x1234, for example as a result of shortening the extension node and
+    /// creating a branch node, the subtrie path is updated to 0x1234.
+    fn get_or_create_lower_subtrie(
```

Comment:
So this isn't something I accounted for in either `reveal_node` or `remove_leaf`. I will need to update `remove_leaf` accordingly, but I think if you incorporated this logic into the existing `lower_subtrie_for_path` and `subtrie_for_path` methods it would fix `reveal_node` without any other changes being necessary, and there wouldn't need to be this extra method.

Reply:
This is basically what I'm thinking, for reference: https://github.com/paradigmxyz/reth/commit/fb9f1570495d237d0f2d8543a029d6d619ee2694

I'm going to work on fixing remove_leaf in that branch, if @Rjected if you wanted to use those changes I could rebase off you later or vice-versa

Reply:
https://github.com/paradigmxyz/reth/pull/17143 is merged so the updated `lower_subtrie_for_path` should be available for you to use here

Reply:
Using the new subtrie for path method now